README.md

```diff
@@ -1,3 +1,95 @@
-# storey
+# Storey
 
 [![CI](https://github.com/v3io/storey/workflows/CI/badge.svg)](https://github.com/v3io/storey/actions?query=workflow%3ACI)
+
+Storey is an asynchronous flow library, for real time event processing and feature extraction.
+
+#### In This Document
+
+- [API Walkthrough](#api-walkthrough)
+- [Usage Examples](#examples)
+
+
+<a id="api-walkthrough"></a>
+## API Walkthrough
+A Storey flow consist of steps linked together by the `build_flow` function, each doing it's designated work.
+
+### Supported Steps
+#### Input Steps
+* `Source` - 
+* `AsyncSource` - 
+* `ReadCSV` - 
+
+#### Processing Steps
+* `Filter` - 
+* `Map` -  
+* `FlatMap` -
+* `MapWithState` -
+* `Batch(max_events, timeout)` - Batches multiple events and emits one event, containing all the batch data, when reaching `max_events` or `timeout` has passed since the first event in the batch.
+* `Choice` - 
+* `JoinWithV3IOTable` - 
+* `JoinWithHttp` - 
+* `AggregateByKey(aggregations,cache, key=None, emit_policy=EmitEveryEvent(), augmentation_fn=None)` - Aggregates data in the event according to the requested aggregates and time windows.
+This step will aggregate the data into the cache object provided, for later persistence and outputs an event which is enriched with the requested aggregation features.
+* `QueryAggregationByKey(aggregations,cache, key=None, emit_policy=EmitEveryEvent(), augmentation_fn=None)` - Similar to to `AggregateByKey`, but this step is for serving only and does not aggregate the event.
+* `Persist(cache)` - Persists the data for the relevant key to the associated storage.
+
+#### Output Steps
+* `Complete` -  
+* `Reduce` -
+* `WriteToV3IOStream` - 
+
+
+<a id="examples"></a>
+## Usage Examples
+
+### Using Aggregates
+The following example reads user data, creates features using Storey's aggregates, persist the data to V3IO and emits the event containing the features to a V3IO Steam for further processing.
+
+```python
+from storey import build_flow, Source, Cache, V3ioDriver, AggregateByKey, FieldAggregator, Persist
+from storey.dtypes import SlidingWindows
+
+v3io_web_api = '****'
+v3io_acceess_key = '****'
+cache = Cache('/bigdata/my_features', V3ioDriver(v3io_web_api, v3io_acceess_key))
```

Comment:
I guess we give valid examples here rather than ****
Something like `https://webapi.change-me.com` and a valid access key example.